src/system/SystemPacketBuffer.h

```diff
@@ -138,6 +138,18 @@ class DLL_EXPORT PacketBuffer : private pbuf
      */
     static constexpr size_t kMaxSize = kMaxSizeWithoutReserve - kDefaultHeaderReserve;
 
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT
+    static constexpr size_t kLargeBufMaxSizeWithoutReserve = CHIP_SYSTEM_CONFIG_MAX_LARGE_BUFFER_SIZE_BYTES;
+
+    static constexpr size_t kLargeBufMaxSize = kLargeBufMaxSizeWithoutReserve - kDefaultHeaderReserve;
+#endif // INET_CONFIG_ENABLE_TCP_ENDPOINT
+
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT
+    static constexpr size_t kMaxAllocSize = kLargeBufMaxSizeWithoutReserve;
+#else
+    static constexpr size_t kMaxAllocSize          = kMaxSizeWithoutReserve;
```

Comment:
I find this confusing: we still have kMaxSizeWithoutReserve however then kMaxSizeWithoutReserve is not used much except in some asserts and in computing these things. Why do we keep the asserts then?

These sizes need some form of diagram/explanation. I do not know what a `reserve` is and I do not understand the difference between "large buffer" vs other.


https://github.com/project-chip/connectedhomeip/blob/master/src/system/SystemPacketBuffer.h#L94 has a diagram in terms of `Member()` calls ... can we have something similar explaining these constants, probably closer to this area? I cannot seem to just read the PR and see that everything is fine and because of the renaming of kMaxSizeWithoutReserve to kMaxAllocSize I worry that we may change behavior in unintended ways.

Reply:
kMaxSizeWithoutReserve is used as the max allocation size for MRP based logic at several places, including Test code. The reserve space is to contain the encapsulation header fields(Matter message and exchange header). It is defined in detail in SystemConfig.h
The renaming to kMaxAllocSize as a unified constant actually aligns with the diagram in https://github.com/project-chip/connectedhomeip/blob/master/src/system/SystemPacketBuffer.h#L94, which is why I chose it. It is, essentially, the max of AllocSize() that encompasses both Reserve and Data space in the buffer.